src/io/linux.zig

```diff
@@ -29,6 +30,33 @@ pub const IO = struct {
     ios_queued: u64 = 0,
     ios_in_kernel: u64 = 0,
 
+    /// The head of a doubly-linked list of all operations that are:
+    /// - in the submission queue, or
+    /// - in the kernel, or
+    /// - in the completion queue, or
+    /// - in the `completed` list (excluding zero-duration timeouts)
+    awaiting: ?*Completion = null,
```

Comment:
Consider extracting the intrusive doubly-linked list into a separate data structure, a-la `FIFO`